Feat/auth kakao

ehgk4245 opened the pull request:
## 긴급 처리 사유

## 📂 작업 내용

closes #34 

- [x] 카카오 소셜 로그인 구현

## 💡 자세한 설명

- spring security .anyRequest().authenticated() 설정 (로그인 필요없는 경로 security 설정 필요)
- 소셜 로그인 시 자체 회원가입된 email이 있을 경우 계정 통합
- 소셜 로그인으로 가입 시 nickname = kakao + 난수(10자리) 초기 지정, password = null, role = UNASSIGNED
- 로그인 성공 페이지에서 role 이 unassigned 일 경우 role 선택 페이지로 이동
- role 선택 시 api 호출 및 디비 반영 , 홈으로 라우팅
- 소셜 로그인 흐름
    - 로그인 페이지에 카카오 로그인 버튼 클릭 -> 백엔드 엔드포인트 호출
    - 백엔드에서 카카오 소셜 로그인 url 리다이렉트 응답
    - 브라우저에서 카카오 서버로 이메일, 패스워드 전달
    - 인증 성공 시 카카오 서버 에서 백엔드 서버로 인증 코드 전달
    - Spring Security에서 인증 코드로 카카오 서버로 부터 사용자 정보 조회까지 내부처리
    - 백엔드 서버에서 사용자 정보 데이터 가공 및 디비 반영 후 응답 쿠키로 RefreshToken 전달 및 next.js 로그인 성공 페이지로 리다이렉트 응답
    - 성공 페이지에서 백엔드 서버에 토큰 재발급 api 호출 후 응답 인증 헤더를 로컬에 저장함으로써 로그인

## 📗 참고 자료 & 구현 결과 (선택)

## 📢 리뷰 요구 사항 (선택)

## 🚩 후속 작업 (선택)
 - 이메일 및 닉네임 중복 확인 기능, 이메일 인증 기능, 비밀번호 재설정 기능 추가예정
## ✅ 셀프 체크리스트

- [x] PR 제목을 형식에 맞게 작성했나요?
- [x] 브랜치 전략에 맞는 브랜치에 PR을 올리고 있나요? (`main`이 아닙니다.)
- [x] 이슈는 close 했나요?
- [x] Reviewers, Labels를 등록했나요?
- [x] 작업 도중 문서 수정이 필요한 경우 잘 수정했나요?
- [x] 테스트는 잘 통과했나요?
- [x] 불필요한 코드는 제거했나요?


<!-- This is an auto-generated comment: release notes by coderabbit.ai -->
## Summary by CodeRabbit

* **New Features**
  * Kakao OAuth login with a success flow that refreshes tokens, loads your profile, and redirects users with no role to a role selection screen.
  * Role selection UI to choose Freelancer or Client; chosen role is applied immediately and reflected in your profile.
  * Profile images now supported and shown in user profiles.

* **Improvements**
  * Logout now redirects to the login page.
  * App uses environment-driven API base URL and recognizes an "unassigned" role state.

* **Documentation**
  * Grouped API documentation enabled.
<!-- end of auto-generated comment: release notes by coderabbit.ai -->

ehgk4245 committed:
♻️ refactor : api 요청 url 하드코딩 -> 환경변수 사용

ehgk4245 committed:
:recycle: refactor : 필터 생성방식 new 키워드 생성 -> 빈 등록 방식으로 변경

ehgk4245 committed:
:recycle: refactor : 필터 생성방식 new 키워드 생성 -> 빈 등록 방식으로 변경

ehgk4245 committed:
✨ feat : 카카오 로그인 기능 추가

ehgk4245 committed:
♻️ refactor : 필요없는 import 제거

ehgk4245 committed:
:recycle: refactor : 개인 설정 제거

ehgk4245 committed:
♻️ refactor : POST 요청 파라미터 전달 -> 바디 전달 변경

ehgk4245 committed:
:recycle: refactor : 사용 안하는 import 제거

ehgk4245 committed:
:recycle: refactor : 요청 DTO 검증 추가

geun-00 reviewed (approved):
깔끔하게 잘 구현하신 것 같습니다! 수고하셨습니다 👍

ehgk4245 committed:
Merge branch 'main' into feat/auth-kakao

ehgk4245 committed:
:sparkles: feat : 소셜 로그인 임시닉네임 난수 설정 방식 변경

# Conflicts:
#	backend/src/main/java/org/team4/project/domain/member/service/MemberService.java

chan808 reviewed (approved):
고생하셨습니다! 병합되면 테스트하면서 바로 사용해보고 싶네요

ehgk4245 committed:
:bug: fix : 병합 중 생긴 컴파일 오류 수정

dbjoung reviewed (approved):
고생하셨습니다..! 덕분에 편하게 로그인 정보 끌어올 수 있을 것 같습니다 : D

ehgk4245 committed:
:recycle: refactor : oauth2 kakao client-id 환경변수로 관리

ehgk4245 committed:
:bug: fix : 역할 선택 api 연동 버그 수정

ehgk4245 committed:
:sparkles: feat : Member 엔티티 profileImageUrl 추가, 프론트 반영
